src/pudl/transform/ferc1.py

```diff
@@ -853,6 +853,632 @@ def strip_col_of_non_ints(self, col: pd.Series) -> pd.DataFrame:
+        ] = "hydro"
+
+        return df
+
+    def associate_notes_with_values(self, df: pd.DataFrame) -> pd.DataFrame:
+        """Use footnotes to map string and ferc license to value rows.
+
+        There are many utilities that report a bunch of note rows at the bottom of their
+        yearly entry. These note rows often pertain directly to specific plant rows
+        above. Sometimes, the notes and their respective plant rows are connected by a
+        footnote such as (a) or (1) etc. This function finds those footnotes, associates
+        the "note" version with the regular value row, maps the note content from the
+        note row into a new note column that's associated with the value row, also maps
+        any ferc license extracted from this note column up to the value row it
+        references.
+        """
+        logger.info(
+            f"{self.table_id.value}: Mapping notes and ferc license from footnotes"
+        )
+
+        def associate_notes_with_values_group(group):
+            """Map footnotes within a given utility year group.
+
+            Because different utilities may use the same footnotes or the same utility
+            could reuse footnotes each year, we must do the footnote association within
+            utility-year groups.
+            """
+            regular_row = group["row_type"].isna()
+            has_note = group["row_type"] == "note"
+            # has_footnote = group.plant_name_ferc1.str.contains(footnote_pattern)
+
+            # Shorten execution time by only looking at groups with discernable footnotes
+            if group.footnote.any():
+
+                # Make a df that combines notes and ferc license with the same footnote
+                footnote_df = (
+                    group[has_note]
+                    .groupby("footnote")
+                    .agg({"plant_name_ferc1": ", ".join, "license_id_ferc1": "first"})
+                    .rename(columns={"plant_name_ferc1": "notes"})
+                )
+
+                # Map these new license and note values onto the original df
+                updated_ferc_license_col = group.footnote.map(
+                    footnote_df["license_id_ferc1"]
+                )
+                notes_col = group.footnote.map(footnote_df["notes"])
+                # We update the ferc lic col because some were already there from the
+                # plant name extraction. However, we want to override with the notes
+                # ferc licenses because they are more likely to be accurate.
+                group.license_id_ferc1.update(updated_ferc_license_col)
+                group.loc[regular_row, "notes"] = notes_col
+
+            return group
+
+        footnote_pattern = r"(\(\d?[a-z]?[A-Z]?\))"
+        df["notes"] = pd.NA
+        df["footnote"] = pd.NA
+        # Create new footnote column
+        df.loc[:, "footnote"] = df.plant_name_ferc1.str.extract(
+            footnote_pattern, expand=False
+        )
```

Comment:
do you need both the null version and the extract version of making this column?

Reply:
don't think so!